### PR TITLE
products page added

### PR DESCRIPTION
**Pull request by sashaboi**
Hello add reviewers ! Here's a preview of what was made. 
you can access the new page directly here : https://deploy-preview-3--spontaneous-longma-8d9e2f.netlify.app/products
screencapture-localhost-3000-products-2022-03-29-17_59_41

An axios call is made in the context file and a reducer is used to supply data to the products page. similarly an axios call is made to the category api in the product file itself, and categories are displayed on the products page. 

Also added a custom navbar feature which tells the navbar which version to render.

Will i remove console.log files from the code base if asked ? : NO ,as i am using them to track the progress of the command.

What else is left ? 
other features
responsiveness, 
exporting reducer function elsewhere to make cleaner code
use of variables in css ( always confused as to why do this , as vs code gives very easy drop downs, but i will do this)

Please suggest changes in structuring, logic and anything else you wish to suggest, thanks !
Onkar Deshpande

### CODE REVIEW

File: src/pages/components/navbar/Navbar.jsx [hunk 1]
```diff
@@ -1,11 +1,14 @@
 import React from 'react'
 import {Link } from "react-router-dom"
+
 import '../navbar/navbar.css'
-const Navbar = () => {
+const Navbar = (mode) => {
+  console.log(mode);
```
Comment (hrishikeshmane) [inline src/pages/components/navbar/Navbar.jsx:6]: Avoid pushing console logs to the repo

**Comment by Pratham82** [comment]
image

UI suggestions:
- From the UI standpoint, the badge would look much better if it's inside the card itself
- The weight badge should be with rounded corners with considering the consistency
- Price can be shown with just text instead of badge
- There should be more margin above add to cart button

Apart from this whole functionality is good and feels snappy.

**Review by Pratham82 — APPROVED** [review]

File: src/pages/context/cartcontext.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+          
+        },
+        (error)=>{
+          console.log(error);
+          
+        })
+    
+        
+    
+      },[])
```
Comment (hrishikeshmane) [inline src/pages/context/cartcontext.js:28]: suggestion: use async await with IIFE

File: src/pages/context/cartcontext.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    
+    const reducerfunc = (state,action)=>{
```
Comment (hrishikeshmane) [inline src/pages/context/cartcontext.js:30]: suggestion: move reducer to a different file.

**Review by hrishikeshmane — APPROVED** [review]
Looks good except for the console logs. Use prettier to format the code before pushing.

File: src/pages/components/navbar/Navbar.jsx (hunk 1 above)
Comment (sashaboi) [reply src/pages/components/navbar/Navbar.jsx:6]: noted, will make sure i run prettier. need the logs for future work , so going to keep them till version 1.0
thanks

File: src/pages/context/cartcontext.js (hunk 2 above)
Comment (sashaboi) [reply src/pages/context/cartcontext.js:28]: Not very familiar with that syntax, so continuing with axios

File: src/pages/context/cartcontext.js (hunk 3 above)
Comment (sashaboi) [reply src/pages/context/cartcontext.js:30]: its in the todo list , want to get a working version 1.0 then make aesthetic changes. will put a pin in this an export before submission.

**Pull request merged**